src/quo2/components/wallet/transaction_progress/view.cljs

```diff
@@ -0,0 +1,306 @@
+                                                              (colors/custom-color :blue 50)
+                                                              (colors/custom-color :blue 60)
+                                                              quo.theme))
+                        :else                         (assoc {:style style/progress-box-arbitrum-abs}
+                                                             :background-color
+                                                             (colors/theme-colors colors/white
+                                                                                  colors/neutral-5
+                                                                                  colors/neutral-70)))]
+        box-style)
+      :align-self "flex-end"
+      :border-color
+      (colors/theme-colors colors/white colors/neutral-10 colors/neutral-80))]]])
+
+(defn render-text
+  [title override-theme &
+   {:keys [typography weight size style]
+    :or   {typography :main-semibold
+           weight     :semi-bold
+           size       :paragraph-1
+           style      (style/title override-theme)}}]
+  [text/text
+   {:typography          typography
+    :accessibility-label :title-name-text
+    :ellipsize-mode      :tail
+    :style               style
+    :override-theme      override-theme
+    :number-of-lines     1
+    :weight              weight
+    :size                size}
+   title])
+
+(defn network-type-text
+  [network-type network-state]
+  (cond
+    (and (= network-state :sending) (= network-type :arbitrum))     "Confirmed on"
+    (or (= network-state :sending) (= network-state :pending))      "Pending on"
+    (or (= network-state :confirmed) (= network-state :finalising)) "Confirmed on"
+    (= network-state :finalized)                                    "Finalized on"
+    (= network-state :error)                                        "Failed on"))
+
+(defn steps-text
+  [network-type network-state]
+  (cond
+    (and (= network-type :mainnet)
+         (not= network-state :finalized)
+         (not= network-state :error))       (str (if (< (@app-state :counter) 4)
+                                                   (@app-state :counter)
+                                                   "4")
+                                                 "/4")
+    (= network-state :finalized)            "Epoch 181,329"
+    (and (= network-type :mainnet)
+         (= network-state :error))          "0/4"
+    (and (not= network-type :mainnet)
+         (or (= network-state :finalising)
+             (= network-state :confirmed))) "1/1"
+    (and (= network-type :arbitrum)
+         (= network-state :sending))        "1/1"
+    (not= network-type :mainnet)            "0/1"))
+
+(defn get-status-icon
+  [network-type network-state]
+  (cond
+    (and (= network-type :arbitrum)
+         (= network-state :sending))   ["positive-state"
```

Comment:
These strings `positive-state`, `pending-state` are references to icons, so they should be keywords with the `:i` qualification, like `:i/positive-state`.